flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlListState.java

```diff
@@ -73,11 +72,14 @@ public void addAll(List<T> values) throws Exception {
 		return () -> new IteratorWithCleanup(finalResult.iterator());
 	}
 
-	private void updateTs(List<TtlValue<T>> ttlValue) throws Exception {
-		List<TtlValue<T>> unexpiredWithUpdatedTs = ttlValue.stream()
-			.filter(v -> !expired(v))
-			.map(this::rewrapWithNewTs)
-			.collect(Collectors.toList());
+	private void updateTs(List<TtlValue<T>> ttlValues) throws Exception {
+		List<TtlValue<T>> unexpiredWithUpdatedTs = new ArrayList<>();
```

Comment:
Would it make sense to initialize the capacity of this `ArrayList` to `ttlValues.size`? That way we would save resizing work.